Update 02-REPL.md (#20)
carpentries-incubator/julia-novice: _episodes/02-REPL.md

{: .output}

That works as expected.

> It is also possible to ~~bind~~ allot names to values via the assignment operator `=`, which makes it easier to refer to them later on.

Reviewer: Actually creating a binding is the official terminology for this.
Do you think allot is easier to understand for newcomers?

Author: You are right. The technical term is variable binding.
My suggestion:

Variable binding are done using assignment operator `=`. Allotting names to variables makes it easier to refer to them later on.

____________
I use the term allot often in my teaching. It has a hidden message, that the name once chosen is reserved.

Reviewer: >  that the name once chosen is reserved.

But that isn't true in julia code. You can bind the same name to different values at different points in the code.
Actually I would avoid the term `variable` since it gives the wrong mental model of "the value is bound to the name" which is the other way around.
Which makes it hard to understand something like
```julia
a = [1,2]
b = a
b[2] = 3
print(a) # gives [1, 3]
```

Author: I quickly searched for the definition on Julia and quote it:

" A variable, in Julia, is a name associated (or bound) to a value. It's useful when you want to store a value (that you obtained after some math, for example) for later use."
https://docs.julialang.org/en/v1/manual/variables/

Albeit variable bind makes a lot of sense now. We should perhaps use this definition.

Reviewer: ```suggestion
It is also possible to create a variable by binding a name to a value via the assignment operator `=`, which makes it easier to refer to them later on.
```
Are you okay with that?

Author: +1

## Unicode

> In julia ~~also~~ unicode characters are also allowed as variables like `α = 2`.
> ~~Where~~ Unicode characters can be entered by a backslash followed by their LaTeX-name and then pressing <kbd>tab</kbd> (in this case `\alpha`<kbd>tab</kbd>).

Reviewer: ```suggestion
In julia unicode characters are also allowed as variables like `α = 2`.
Unicode characters can be entered by a backslash followed by their LaTeX-name and then pressing <kbd>tab</kbd> (in this case `\alpha`<kbd>tab</kbd>).
```

She gets back to normal mode by pressing backspace.

Another useful mode is the shell mode that can be entered by pressing <kbd>;</kbd>.
The prompt has now changed to shell.

> It can be used to ~~use~~ issue commands of the underlying shell, but don't confuse it with an actual shell: Special shell syntax like piping won't work.

Reviewer: ```suggestion
It can be used to issue commands of the underlying shell, but don't confuse it with an actual shell: Special shell syntax like piping won't work.
```

> ## Hello shell mode
> Use the shell mode to start nano and save your first `.jl` file
{: .challenge}

> Finally there is the package mode that is ~~enetered~~ entered with <kbd>]</kbd> which is used for package management, which will be covered later on.

Reviewer: ```suggestion
Finally there is the package mode that is entered with <kbd>]</kbd> which is used for package management, which will be covered later on.
```
